pydantic_ai_slim/pydantic_ai/_tool_manager.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from dataclasses import dataclass, replace
+from typing import Any, Generic
+
+from pydantic import ValidationError
+
+from pydantic_ai.output import DeferredToolCalls
+
+from . import messages as _messages
+from ._run_context import AgentDepsT, RunContext
+from .exceptions import ModelRetry, ToolRetryError, UnexpectedModelBehavior
+from .messages import ToolCallPart
+from .tools import ToolDefinition
+from .toolsets.abstract import AbstractToolset, ToolsetTool
+
+
+@dataclass
+class ToolManager(Generic[AgentDepsT]):
+    """TODO: Update docstring -- A toolset that caches the wrapped toolset's tool definitions for a specific run step and handles retries."""
+
+    ctx: RunContext[AgentDepsT]
+    toolset: AbstractToolset[AgentDepsT]
+    tools: dict[str, ToolsetTool[AgentDepsT]]
```

Comment:
I think it merits having a docstring here explaining that this is a cached result of listing the toolset's tools. I mean I see you probably intend to update docstring/etc. here but just making a concrete suggestion.